pgx/src/datum/array.rs

```diff
@@ -331,23 +331,6 @@ impl<T: FromDatum> FromDatum for Vec<T> {
     }
 }
 
-impl<T: FromDatum> FromDatum for Vec<Option<T>> {
```

Comment:
Why did this go away?  This is for supporting postgres `ARRAY[]`s that might contain `NULL` elements

Reply:
If I re-add this and remove that conflicting implementation I get a lot of type failures, eg

```rust
error[E0277]: the trait bound `std::option::Option<heap_tuple::PgHeapTuple<'static, pgbox::AllocatedByRust>>: datum::PostgresType` is not satisfied
   --> pgx/src/lib.rs:263:26
    |
263 |             TypeId::of::<Array<Option<PgHeapTuple<'static, AllocatedByRust>>>>(),
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `datum::PostgresType` is not implemented for `std::option::Option<heap_tuple::PgHeapTuple<'static, pgbox::AllocatedByRust>>`
    |
note: required because of the requirements on the impl of `from::FromDatum` for `std::option::Option<heap_tuple::PgHeapTuple<'static, pgbox::AllocatedByRust>>`
   --> pgx/src/datum/varlena.rs:344:14
```